src/lightning/fabric/utilities/seed.py

```diff
@@ -91,24 +97,26 @@ def pl_worker_init_function(worker_id: int, rank: Optional[int] = None) -> None:
     log.debug(
         f"Initializing random number generators of process {global_rank} worker {worker_id} with base seed {base_seed}"
     )
-    ss = np.random.SeedSequence([base_seed, worker_id, global_rank])
-    # use 128 bits (4 x 32-bit words)
-    np.random.seed(ss.generate_state(4))
-    # Spawn distinct SeedSequences for the PyTorch PRNG and the stdlib random module
-    torch_ss, stdlib_ss = ss.spawn(2)
-    torch.manual_seed(torch_ss.generate_state(1, dtype=np.uint64)[0])
-    # use 128 bits expressed as an integer
-    stdlib_seed = (stdlib_ss.generate_state(2, dtype=np.uint64).astype(object) * [1 << 64, 1]).sum()
+    if _NUMPY_AVAILABLE:
+        ss = np.random.SeedSequence([base_seed, worker_id, global_rank])
+        # use 128 bits (4 x 32-bit words)
+        np.random.seed(ss.generate_state(4))
+        # Spawn distinct SeedSequences for the PyTorch PRNG and the stdlib random module
+        torch_ss, stdlib_ss = ss.spawn(2)
+        torch.manual_seed(torch_ss.generate_state(1, dtype=np.uint64)[0])
+        # use 128 bits expressed as an integer
+        stdlib_seed = (stdlib_ss.generate_state(2, dtype=np.uint64).astype(object) * [1 << 64, 1]).sum()
     random.seed(stdlib_seed)
```

Comment:
This here wouldn't be defined if numpy wasn't available. Also we still need to call torch.manual_seed() with a seed even if numpy is not available. This is a tricky part that we need to rewrite. Let me think about this for a bit to find a solution

Reply:
```
if _NUMPY_AVAILABLE:
        ss = np.random.SeedSequence([base_seed, worker_id, global_rank])
        np.random.seed(ss.generate_state(4))
        torch_ss, stdlib_ss = ss.spawn(2)
        torch.manual_seed(torch_ss.generate_state(1, dtype=np.uint64)[0])
        stdlib_seed = (stdlib_ss.generate_state(2, dtype=np.uint64).astype(object) * [1 << 64, 1]).sum()
        random.seed(stdlib_seed)
    else:
        torch.manual_seed(base_seed)
        random.seed(base_seed)
```
@awaelchli  can you Please check this one good enough ??

Reply:
It's very close, but we will need an implementation for generating the random state like we did before, but without numpy (with torch instead). I can provide that implementation, or you can try it yourself too of course. I'll reserve some time to work on this and get back to you.